day-04/part-1/badouralix.py

```diff
@@ -0,0 +1,17 @@
+from tool.runners.python import SubmissionPy
+
+
+class BadouralixSubmission(SubmissionPy):
+    def run(self, s):
+        """
+        :param s: input in string format
+        :return: solution flag
+        """
+        result = 0
+
+        for line in s.split("\n\n"):
+            keyvalues = set(line.replace("\n", " ").replace(":", " ").split())
+            if {"byr", "iyr", "eyr", "hgt", "hcl", "ecl", "pid"} <= keyvalues:
```

Comment:
This looks optimistic, the expected keys can be in the values ;-)

Reply:
Fair enough :joy: Fixed in https://github.com/david-ds/adventofcode-2020/pull/80
The dataset generator guarantees that it never happens though :innocent: